visualizer/parseTree.js

```diff
@@ -645,6 +645,10 @@
     refreshParseTree(trace, true);
   });
 
+  ohmEditor.addListener('change:action', function(trace) {
```

Comment:
I don't think you need this. Instead, just call `ohmEditor.parseTree.refresh()` from the semantics editor when the selected action is changed.